pkg/operation/seed/seed.go

```diff
@@ -849,3 +863,22 @@ func GetWildcardCertificate(ctx context.Context, c client.Client) (*corev1.Secre
 	}
 	return nil, nil
 }
+
+// determineClusterIdentity determines the identity of a cluster, in cases where the identity was
+// created manually or the Seed was created as Shoot, and later registered as Seed and already has
+// an identity, it should not be changed.
+func determineClusterIdentity(ctx context.Context, c client.Client) (string, error) {
+	clusterIdentity := &corev1.ConfigMap{}
+	if err := c.Get(ctx, kutil.Key(metav1.NamespaceSystem, v1beta1constants.ClusterIdentity), clusterIdentity); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return "", err
+		}
+
+		gardenNamespace := &corev1.Namespace{}
+		if err := c.Get(ctx, kutil.Key(v1beta1constants.GardenNamespace), gardenNamespace); err != nil {
+			return "", err
+		}
+		return string(gardenNamespace.UID), nil
```

Comment:
I don't understand why do we set the Seed cluster identity to the garden namespace UID if the configmap is not found? Shouldn't we generate it in a manner similar to the shoot cluster identity? Or vice versa, why do we bother generating the Shoot cluster identity in the way above instead of just using an UID there as well (see my comment above)? Why the difference between Seed and Shoot?

Reply:
Hm, good point. Doesn't this also mean that every seed will have the same identity (=uid of garden NS)?

Reply:
No, we are using the Garden ns in the Seed

Reply:
As the proposal goes when cluster identity is missing (configMap) we generate it using the `Namespace` UID. Maybe i have to change it to use the `kube-system` namespace instead of the garden namespace, because it could be missing in the Seed.

Reply:
Yes, I also think `kube-system` is better, since Seeds (unlike Shoots) don't have own UIDs. How about including things like the seed name, and perhaps the garden identity in the Seed identity as well, similarly to the Shoots?